src/components/utils/test/posix_thread_test.cc

```diff
@@ -41,287 +43,466 @@ namespace utils_test {
   // Arrange
   threads::Thread* thread = NULL;
   TestThreadDelegate* threadDelegate = new TestThreadDelegate();
-  AutoLock test_lock(test_mutex_);
-  // Create thread
-  ASSERT_NO_THROW(thread = CreateThread(threadName, threadDelegate));
-  // Start thread with following options (Stack size = 32768 & thread is
-  // joinable)
-  thread->start(threads::ThreadOptions(MyStackSize));
-  // Check thread is joinable
-  EXPECT_TRUE(thread->is_joinable());
-  // Check thread stack size is 32768
-  EXPECT_EQ(MyStackSize, thread->stack_size());
-  cond_var_.WaitFor(test_lock, 10000);
+
+  {
+    AutoLock test_lock(*threadDelegate);
+    // Create thread
+    ASSERT_NO_THROW(thread = CreateThread(kThreadName, threadDelegate));
+    // Start thread with following options (Stack size = 32768 & thread is
+    // joinable)
+    thread->Start(threads::ThreadOptions(kStackSize));
+    // Check thread is joinable
+    EXPECT_TRUE(thread->IsJoinable());
+    // Check thread stack size is 32768
+    EXPECT_EQ(kStackSize, thread->StackSize());
+    threadDelegate->WaitFor(test_lock, 10000);
+  }
+
   EXPECT_TRUE(threadDelegate->check_value());
-  DeleteThread(thread);
+
   delete threadDelegate;
-  EXPECT_EQ(NULL, thread->delegate());
+  EXPECT_EQ(NULL, thread->GetDelegate());
+  DeleteThread(thread);
 }
 
-TEST(
+TEST_F(
     PosixThreadTest,
     StartCreatedThreadWithDefaultOptions_ExpectZeroStackAndJoinableThreadStarted) {
   // Arrange
   threads::Thread* thread = NULL;
   TestThreadDelegate* threadDelegate = new TestThreadDelegate();
-  AutoLock test_lock(test_mutex_);
-  // Create thread
-  ASSERT_NO_THROW(thread = CreateThread(threadName, threadDelegate));
-  // Start thread with default options (Stack size = 0 & thread is joinable)
-  thread->start(threads::ThreadOptions());
-  // Check thread is joinable
-  EXPECT_TRUE(thread->is_joinable());
-  // Check thread stack size is minimum value. Stack can not be 0
-  EXPECT_EQ(Thread::kMinStackSize, thread->stack_size());
-  cond_var_.WaitFor(test_lock, 10000);
+
+  {
+    AutoLock test_lock(*threadDelegate);
+    // Create thread
+    ASSERT_NO_THROW(thread = CreateThread(kThreadName, threadDelegate));
+    // Start thread with default options (Stack size = 0 & thread is joinable)
+    thread->Start(threads::ThreadOptions());
+    // Check thread is joinable
+    EXPECT_TRUE(thread->IsJoinable());
+    // Check thread stack size is minimum value. Stack can not be 0
+    EXPECT_EQ(Thread::kMinStackSize, thread->StackSize());
+    threadDelegate->WaitFor(test_lock, 10000);
+  }
+
   EXPECT_TRUE(threadDelegate->check_value());
-  DeleteThread(thread);
+
   delete threadDelegate;
-  EXPECT_EQ(NULL, thread->delegate());
+  EXPECT_EQ(NULL, thread->GetDelegate());
+  DeleteThread(thread);
 }
 
-TEST(
+TEST_F(
     PosixThreadTest,
-    StartThreadWithZeroStackAndDetached_ExpectMinimumStackAndDetachedThreadStarted) {
+    DISABLED_StartThreadWithZeroStackAndDetached_ExpectMinimumStackAndDetachedThreadStarted) {
```

Comment:
@SNiukalov why it's disabled?

Reply:
Propose to remove detached state, and make warning in code, before creating thread, that detached is not supported

Reply:
@LitvinenkoIra @LuxoftAKutsan 
I leaved this test  https://github.com/LuxoftSDL/sdl_core/blob/ca62978ee665d27ecfceffac8caf96ce740ecb51/src/components/utils/test/posix_thread_test.cc#L196-L212, because with new implementation it is correct working, but added warning in code:
https://github.com/LuxoftSDL/sdl_core/blob/2b36c78608298cb6602a41956a61db00ae2ef927/src/components/utils/src/threads/thread_posix.cc#L349-L352